sklearn/ensemble/voting_classifier.py

```diff
@@ -0,0 +1,206 @@
+    voting : str, {'hard', 'soft'} (default='hard')
+      If 'hard', uses predicted class labels for majority rule voting.
+      Else if 'soft', predicts the class label based on the argmax of 
+      the sums of predict probalities.
+    
+    weights : array-like, shape = [n_classifiers], optional (default=`None`)
+      Sequence of weights (`float` or `int`) that are multiplied with the 
+      predicted class probabilities before averaging if `voting='soft'` 
+      Uses uniform weights if `None`.
+        
+    Attributes
+    ----------
+    classes_ : array-like, shape = [n_cl_lab]
+        
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> from sklearn.naive_bayes import GaussianNB 
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> np.random.seed(123)
+    >>> clf1 = LogisticRegression()
+    >>> clf2 = RandomForestClassifier()
+    >>> clf3 = GaussianNB()
+    >>> X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]])
+    >>> y = np.array([1, 1, 1, 2, 2, 2])
+    >>> eclf1 = VotingClassifier(estimators=[clf1, clf2, clf3], voting='hard')
+    >>> eclf1 = eclf1.fit(X, y)
+    >>> print(eclf1.predict(X))
+    [1 1 1 2 2 2]
+    >>> eclf2 = VotingClassifier(estimators=[clf1, clf2, clf3], voting='soft')
+    >>> eclf2 = eclf2.fit(X, y)
+    >>> print(eclf2.predict(X))
+    [1 1 1 2 2 2]
+    >>> eclf3 = VotingClassifier(estimators=[clf1, clf2, clf3], voting='soft', weights=[2,1,1])
+    >>> eclf3 = eclf3.fit(X, y)
+    >>> print(eclf3.predict(X))
+    [1 1 1 2 2 2]
+    >>>
+      
+    """
+    def __init__(self, estimators, voting='hard', weights=None):
+        self.estimators = estimators
+        self.voting = voting
+        self.weights = weights
+        self.le = LabelEncoder()
+
+    
+    def fit(self, X, y):
+        """ 
+        Fits the estimators.
+        
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+            
+        y : array-like, shape = [n_samples]
+            Target values.
+      
+        Returns
+        -------
+        self : object
+
+        """
+        
+        if self.voting not in ('soft', 'hard'):
+            raise ValueError("voting must be 'soft' or 'hard'; got (voting=%r)"
+                             % voting)
+                                                  
+        if self.weights and len(self.weights) != len(self.estimators):
+            raise ValueError('Number of classifiers and weights must be equal'\
+                         '; got %d weights, %d estimators' 
+                         % (len(self.weights), len(self.estimators)))      
+        
+        for clf in self.estimators:
+            clf.fit(X, y)
+        
+        self.le.fit(y)
+        self.classes_ = np.unique(y)
```

Comment:
I think if the estimator has a label encoder, we use
``python
@property
def classes_(self):
    return self.le_.classes_
``
but please double check.